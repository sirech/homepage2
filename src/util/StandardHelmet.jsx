import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import { siteMetadata } from '../../gatsby-config'

const StandardHelmet = ({ title, description }) => (
  <Helmet
    title={`${title} | ${siteMetadata.title}`}
    meta={[
      { name: 'title', content: `${title} | ${siteMetadata.title}` },
      { name: 'description', content: description },
      { name: 'twitter:card', content: 'summary' },
      { property: 'og:title', content: title },
      { property: 'og:type', content: 'website' },
      {
        property: 'og:description',
        content: description,
      },
    ]}
  />
)

StandardHelmet.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
}

export default StandardHelmet
