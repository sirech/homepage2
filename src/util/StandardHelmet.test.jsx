import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import StandardHelmet from './StandardHelmet'

describe('components', () => {
  describe('StandardHelmet', () => {
    it('renders correctly', () => {
      const component = shallow(
        <StandardHelmet title="the title" description="the description" />
      )
      expect(toJson(component)).toMatchSnapshot()
    })
  })
})
