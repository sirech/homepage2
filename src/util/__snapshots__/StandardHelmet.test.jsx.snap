// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components StandardHelmet renders correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  meta={
    Array [
      Object {
        "content": "the title | Mario Fernandez",
        "name": "title",
      },
      Object {
        "content": "the description",
        "name": "description",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
      Object {
        "content": "the title",
        "property": "og:title",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "the description",
        "property": "og:description",
      },
    ]
  }
  title="the title | Mario Fernandez"
/>
`;
